tests/test_dse.py

```diff
@@ -0,0 +1,295 @@
+from conftest import EVAL
+
+import numpy as np
+import pytest
+from sympy import Eq, Symbol  # noqa
+
+from devito.dse import (clusterize, rewrite, xreplace_constrained, iq_timeinvariant,
+                        iq_timevarying, estimate_cost, temporaries_graph,
+                        common_subexprs_elimination, collect_aliases)
+from devito import Dimension, x, y, z, time, TimeData, clear_cache  # noqa
+from devito.nodes import Expression
+from devito.stencil import Stencil
+from devito.visitors import FindNodes
+from examples.acoustic.Acoustic_codegen import Acoustic_cg
+from examples.containers import IShot
+from examples.seismic import Model
+from examples.source_type import SourceLike
+from examples.tti.tti_example import setup
+from examples.tti.tti_operators import ForwardOperator
+
+
+# Acoustic
+
+
+def run_acoustic_forward(dse=None):
+    # TODO: temporary work around to issue #225 on GitHub
+    clear_cache()
+
+    dimensions = (50, 50, 50)
+    origin = (0., 0., 0.)
+    spacing = (10., 10., 10.)
+    nbpml = 10
+
+    # True velocity
+    true_vp = np.ones(dimensions) + 2.0
+    true_vp[:, :, int(dimensions[0] / 2):int(dimensions[0])] = 4.5
+
+    model = Model(origin, spacing, dimensions, true_vp, nbpml=nbpml)
+
+    # Define seismic data.
+    data = IShot()
+    src = IShot()
+    f0 = .010
+    dt = model.critical_dt
+    t0 = 0.0
+    tn = 250.0
+    nt = int(1+(tn-t0)/dt)
+
+    t = np.linspace(t0, tn, nt)
+    r = (np.pi * f0 * (t - 1./f0))
+    # Source geometry
+    time_series = np.zeros((nt, 1))
+    time_series[:, 0] = (1-2.*r**2)*np.exp(-r**2)
+
+    location = np.zeros((1, 3))
+    location[0, 0] = origin[0] + dimensions[0] * spacing[0] * 0.5
+    location[0, 1] = origin[1] + dimensions[1] * spacing[1] * 0.5
+    location[0, 2] = origin[1] + 2 * spacing[2]
+    src.set_receiver_pos(location)
+    src.set_shape(nt, 1)
+    src.set_traces(time_series)
+
+    # Receiver geometry
+    receiver_coords = np.zeros((101, 3))
+    receiver_coords[:, 0] = np.linspace(0, origin[0] +
+                                        dimensions[0] * spacing[0], num=101)
+    receiver_coords[:, 1] = origin[1] + dimensions[1] * spacing[1] * 0.5
+    receiver_coords[:, 2] = location[0, 1]
+    data.set_receiver_pos(receiver_coords)
+    data.set_shape(nt, 101)
+    acoustic = Acoustic_cg(model, data, src, dse=dse, dle='basic')
+    rec, u, _, _, _ = acoustic.Forward(save=False, dse=dse, dle='basic')
+
+    # FIXME: note that np.copy is necessary because of the broken caching system
+    return np.copy(rec), np.copy(u.data)
```

Comment:
Maybe point to a test that would fail without this? Would be helpful to whoever fixes this later.